client.go

```diff
@@ -177,6 +182,27 @@ func (c *Client) Get(url string, resp interface{}) error {
 	return c.Do(req, nil, resp)
 }
 
+// Post is a convenience method that uses c.Do to issue a POST request to
+// the given URL. If c.BaseURL is configured then url will be treated as
+// being relative to c.BaseURL therefore should not contain a host part.
+// If params is non-nil then it will
+func (c *Client) Post(url string, params, resp interface{}) error {
```

Comment:
As far as I can see that will make the code for my current use case more complex, that doesn't in itself justify not changing the API, but should be considered before we change it.